addons/auth_signup_recaptcha/static/src/signup.js

```diff
@@ -0,0 +1,32 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { ReCaptcha } from "@google_recaptcha/js/recaptcha";
+
+publicWidget.registry.SignupCaptcha = publicWidget.Widget.extend({
+    selector: '.oe_signup_form',
+    events: {
+        'submit': '_onSubmit',
+    },
+
+    init() {
+        this._super(...arguments);
+        this._recaptcha = new ReCaptcha();
+        this.shouldCatch = true;
+    },
+
+    async willStart() {
+        return this._recaptcha.loadLibs();
+    },
+
+    _onSubmit(event) {
+        if(this.shouldCatch) {
+            event.preventDefault()
+            this._recaptcha.getToken('portal_signup').then(tokenCaptcha => {
+                this.$el.append(`<input name="recaptcha_token_response" type="hidden" value="${tokenCaptcha.token}"/>`);
+                this.shouldCatch = false;
+                this.$el.submit();
+            })
+        }
+    },
+})
```

Comment:
You copy/paste the same code twice. I guess three times if I ask you to add the protection to login as well.

Is there no way to factorize the common code ?

It could also be an owl component that we then add in the Qweb templates of login/signup/reset password with a regular back-end view override where we add an `<owl-component/>` node.